common/edl/socket.edl

```diff
@@ -19,9 +19,28 @@
 enclave
 {
 
-    include "openenclave/internal/syscall/netdb.h"
-    include "openenclave/internal/syscall/sys/socket.h"
-    include "openenclave/internal/syscall/types.h"
+
+    // Headers needed for integral types
+    include "openenclave/corelibc/bits/types.h"
+    include "openenclave/bits/edl/syscall_types.h"
+
+    struct oe_sockaddr
+    {
+        oe_sa_family_t sa_family;
+        char sa_data[14];
+    };
+
+    struct oe_addrinfo
+    {
+        int ai_flags;
+        int ai_family;
+        int ai_socktype;
+        int ai_protocol;
+        oe_socklen_t ai_addrlen;
+        struct oe_sockaddr* ai_addr;
```

Comment:
can ai_addr should be annotated as having ai_addrlen be the size of the buffer?
ai_addr can point to a buffer of length up to sizeof(sockaddr_storage)

Reply:
You're right, it can. I opened an issue to evaluate internal EDL structures for what annotation can be added to them: https://github.com/openenclave/openenclave/issues/2760

Reply:
I had to fix a Windows build issue so I went ahead and fixed this in this PR

Reply:
Except you have a typo :)